ext/compressors/iaa/iaa_compress.c

```diff
@@ -0,0 +1,208 @@
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
+/*
+ * We need to include the configuration file to detect whether this extension is being built into
+ * the WiredTiger library; application-loaded compression functions won't need it.
+ */
+#include <wiredtiger_config.h>
+
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+#include <time.h>
+#include "iaaInterface-c.h"
+
+
+#ifdef _MSC_VER
+#define inline __inline
+#endif
+
+/* Local compressor structure. */
+typedef struct {
+    WT_COMPRESSOR compressor; /* Must come first */
+
+    WT_EXTENSION_API *wt_api; /* Extension API */
+} iaa_COMPRESSOR;
+
+/*
+ * iaa_error --
+ *     Output an error message, and return a standard error code.
+ */
+static int
+iaa_error(WT_COMPRESSOR *compressor, WT_SESSION *session, const char *call, int error)
+{
+    WT_EXTENSION_API *wt_api;
+
+    wt_api = ((iaa_COMPRESSOR *)compressor)->wt_api;
+
+    (void)wt_api->err_printf(wt_api, session, "iaa error: %s: %u", call, error);
+    return (WT_ERROR);
+}
+
+/*
+ * iaa_compress --
+ *     WiredTiger iaa compression.
+ */
+static int
+iaa_compress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len,
+  uint8_t *dst, size_t dst_len, size_t *result_lenp, int *compression_failed)
+{
+    uint32_t result_len = doCompressData(compressor, session, src, (uint32_t)src_len, dst, (uint32_t)dst_len);
+    if(result_len > 0) {
+        *compression_failed = 0;
+        *result_lenp = (size_t)result_len;
+        return (0);
+    }
+    *compression_failed = 1;
+    return (iaa_error(compressor, session, "iaa_compress", -1));
+}
+
+/*
+ * iaa_decompress --
+ *     WiredTiger iaa decompression.
+ */
+static int
+iaa_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len,
+  uint8_t *dst, size_t dst_len, size_t *result_lenp)
+{
+    uint32_t result_len = 0;
```

Comment:
Please separate the definition from the initialisation.

Reply:
Hi, etienneptl
OK, we will do the separation.
Thank you!